### DISCUSSION TITLE
Allow embed to read background color from query parameter

### DISCUSSION THREAD

**Discussion by tynandebold** [dawnlabs/carbon#1368] [Q&A] [answered]
I'd love to be able to use a preset theme but override the background color from the embed URL. Is that possible at the current state? It doesn't seem like it is.

### Use Case

We're embedding Carbon iframes with the `Monokai` theme. Our designers want to change the background color to `rgba(21,23,24,100)`. We set it in the URL, but [it doesn't render](https://carbon.now.sh/embed?bg=rgba%2821%2C23%2C24%2C100%29&t=monokai&wt=sharp&l=python&width=660&ds=false&dsyoff=3px&dsblur=13px&wc=true&wa=false&pv=0px&ph=0px&ln=false&fl=1&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false&code=import%2520pandas%2520as%2520pd%250Afrom%2520kedro.pipeline%2520import%2520pipeline%252C%2520node%250A%250Adef%2520clean_data%28weather_df%253A%2520pd.DataFrame%29%2520-%253E%2520pd.DataFrame%253A%250A%2520%2520%2520%2520clean_weather_df%2520%253D%2520weather_df.drop%28%2522temperature%2522%252C%2520axis%253D1%29%250A%2520%2520%2520%2520return%2520clean_weather_df%250A%250Adef%2520join_meteoro_data%28%250A%2520%2520%2520%2520weather_df%253A%2520pd.DataFrame%252C%2520climate_df%253A%2520pd.DataFrame%29%2520-%253E%2520pd.DataFrame%253A%250A%2520%2520%2520%2520weather_climate_df%2520%253D%2520weather_df.merge%28climate_df%252C%2520on%253D%255B%2522country_code%2522%252C%2520%2522area_code%2522%255D%252C%2520how%253D%2522inner%2522%29%250A%2520%2520%2520%2520return%2520weather_climate_df%250A%250Adata_processing_pipeline%2520%253D%2520pipeline%28%250A%2520%2520%2520%2520%255B%250A%2520%2520%2520%2520%2520%2520%2520%2520node%28%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520func%253Dclean_data%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520inputs%253D%2522weather_data%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520outputs%253D%2522clean_weather_data%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%29%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520node%28%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520func%253Djoin_meteoro_data%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520inputs%253D%255B%2522clean_weather_data%2522%252C%2520%2522climate_change_data%2522%255D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520outputs%253D%2522weather_climate_data%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%29%250A%2520%2520%2520%2520%255D%250A%29) with that color.

I'm wondering why that query param is there? Does something else act upon it?

**Comment by mfix22** [comment, marked as answer]
https://carbon.now.sh/embed?bg=red

**Reply by tynandebold** [reply]
Fascinating. So this sets the background color of the container, not the code snippet, which is what we thought it did. We're using this with 0 x and y padding, so when changing the background color nothing changes visually, hence my original question.

What about adding a query param that would override the `.CodeMirror` background color? This is what we were trying to achieve.

**Reply by mfix22** [reply]
Background color is dictated by the highlight theme. So if you create a custom theme, you can customize the background color there.

**Reply by tynandebold** [reply]
Indeed. Though if you create a custom theme and then want to embed that creation, your snippet won't appear as you designed it, since the customization parameters don't travel with the iframe. They're set in Local Storage only. Is that correct?

If so, back to my question from above: what about adding a query param that would override the `.CodeMirror` background color? Happy to look into a PR if you're fine with the idea.

**Reply by mfix22** [reply]
You can save the snippet and then embed that saved snippet, and the highlights will come with it. 

I'd be open to accepting a PR for your suggested idea, so long as it doesn't perturb the current state implementation too much 👍

**Reply by ghost** [reply]
This seems not to work with gists :(  So when I create one via gist and do change the colours etc the url stays like /embed/<gist-id>
